backend/node/genvm/base.py

```diff
@@ -133,9 +134,59 @@ async def run_contract(
     def get_contract_schema(contract_code: str) -> dict:
 
         namespace = {}
-        exec(contract_code, globals(), namespace)
+        try:
+            exec(contract_code, globals(), namespace)
+        except SyntaxError as err:
+            cl, exc, tb = sys.exc_info()
+            
+            return {
+                "exec_error": {
```

Comment:
nit: I think it would be better to create some type for this function which could encapsulate both the happy path and the errors raised

Reply:
@bradleySuira, Also, the return schema of the function should stay the same, adding the errors. Let's move these checks to the code_enforcement and run it like you do on line 189. If code_enforcement returns an error, you can finish this function execution earlier, returning whatever you can return (class_name, etc) plus the errors:

```
contract_schema = {
            "class": class_name,
            "methods": methods,
            "variables": variables,
            "errors": []
        }
```